client/src/pages/SearchPage.js

```diff
@@ -4,42 +4,49 @@ import CategoryList from "../components/CategoryList";
 import NavBar from "../components/NavBar";
 import InfoBox from "../components/InfoBox";
 import { getProductsByTitle } from "../utils/api";
-import { useState, useEffect } from "react";
-import useAsync from "../hooks/useAsync";
+import { useState, useEffect, useCallback } from "react";
 import useFavorites from "../hooks/useFavorites";
+import useDebounce from "../hooks/useDebounce";
+import useAsync from "../hooks/useAsync";
 
 export const SearchPage = () => {
   const { favorites, toggleFavorite } = useFavorites("favorites", []);
-  const [inputValue, setInputValue] = useState("");
-  const { data, error, loading, fetchData } = useAsync(
-    getProductsByTitle,
-    inputValue
-  );
+  const [searchTerm, setSearchTerm] = useState("");
+  const debouncedSearchTerm = useDebounce(searchTerm, 400);
+  const [data, setData] = useState(null);
+  const { loading, error } = useAsync();
+
+  const getData = useCallback(async () => {
+    return await getProductsByTitle(debouncedSearchTerm);
+  }, [debouncedSearchTerm]);
+
+  useEffect(() => {
+    const searchData = async () => {
+      if (debouncedSearchTerm) {
+        setData(await getData());
+      }
+    };
+    searchData();
+  }, [debouncedSearchTerm, getData]);
```

Comment:
A good idea, but not quite what you wanted I think :). It looks like it might work, but some functionality might now be a bit broken.

The problem is, that you have replaced parts of your `useAsync` hook which makes `loading` and `error` redundant. They never change, because you aren't using `fetchData`.  What you want to do is only call `fetchData` if the debounced value changes.

Almost there 🐜